Include/internal/pycore_runtime.h

```diff
@@ -118,25 +120,26 @@ typedef struct pyruntimestate {
 
     struct _Py_unicode_runtime_ids unicode_ids;
 
-    struct _Py_global_objects global_objects;
-    // If anything gets added after global_objects then
-    // _PyRuntimeState_reset() needs to get updated to clear it.
+    struct {
+        struct _Py_global_objects global_objects;
+        struct _is interpreters_main;
+        // XXX Pre-allocate as many objects as possible here.
+    } _preallocated;
```

Comment:
Isn't everything in the runtime state pre-allocated?

Reply:
Almost everything.  There are some locks that we don't pre-allocate.  I'll drop that TODO and add a note about the locks.

Reply:
Could you remove this struct then, it doesn't add anything.

Reply:
This struct statically allocates data that was previously dynamically allocated.  When I said almost everything is pre-allocated, I was including dynamic allocation.  The point of this struct is to group the statically allocated data (which will be accessed via pointer fields in `_PyRuntimeState`).  To me, it makes it more obvious that the data is distinct from the other `_PyRuntimeState` fields.  (In this case, the data is accessed from pointer fields elsewhere in `_PyRuntimeState` rather than directly.)  Is there a disadvantage to this approach?  Should we get rid of the corresponding pointer fields?

FWIW, I considered having a separate static variable for this stuff instead of a sub-struct field but didn't see the point.

Reply:
I'm afraid that doesn't make anything clearer to me.
Please remove these layered structs. They really don't add anything.

Also, you don't need comments explaining how C works. If an object is not a pointer, then it must be allocated at the same time as the enclosing struct.

The above struct could be replaced with:
```C
struct _Py_global_objects global_objects;
struct _is main_interpreter;
```
Which is simpler, and IMO clearer.

Reply:
I've dropped the "_preallocated" sub-struct.